expression/builtin_json_vec.go

```diff
@@ -295,11 +296,67 @@ func (b *builtinJSONRemoveSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Col
 }
 
 func (b *builtinJSONMergeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONMergeSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	nr := input.NumRows()
+	argBuffers := make([]*chunk.Column, len(b.args))
+	var err error
+	for i, arg := range b.args {
+		if argBuffers[i], err = b.bufAllocator.get(types.ETJson, nr); err != nil {
+			return err
+		}
+		defer func(buf *chunk.Column) {
+			b.bufAllocator.put(buf)
+		}(argBuffers[i])
+
+		if err := arg.VecEvalJSON(b.ctx, input, argBuffers[i]); err != nil {
+			return err
+		}
+	}
+
+	jsonValues := make([][]json.BinaryJSON, nr)
+
+	for i := 0; i < nr; i++ {
```

Comment:
Can we merge the 3 loops?

Reply:
the second loop needs to run after the first loop, so does the third loop.

Reply:
Can we refine it like the following? If we want to reduce the cost of `runtime.growslice`, we can check `isNull` and then malloc jsonValues[i]. 
``` go
	result.ReserveJSON(nr)
OUTER:	
	for i := 0; i < nr; i++ {
		for j := 0; j < len(b.args); j++ {
			if argBuffers[j].IsNull(i) {
				result.AppendNull()
				goto OUTER
			}
			jsonValues[i] = append(jsonValues[i], argBuffers[i].GetJSON(j))
		}
		result.AppendJSON(json.MergeBinary(jsonValues[i]))
	}
```

Reply:
I think this approch reduces the locality of memory accesses. In this approch, we have to go through all the argument columns row by row. I will do a benchstat to test it.

Reply:
ok